website/project/views/register.py

```diff
@@ -47,6 +55,231 @@ def node_register_page(auth, node, **kwargs):
     ret.update(_view_project(node, auth, primary=True))
     return ret
 
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_be_public_registration
+def node_registration_retraction_get(auth, node, **kwargs):
+    """Prepares node object for registration retraction page.
+
+    :return: serialized Node to be retracted
+    :raises: 400: BAD_REQUEST if registration already pending retraction
+    """
+
+    if node.pending_retraction:
+        raise HTTPError(http.BAD_REQUEST, data={
+            'message_short': 'Invalid Request',
+            'message_long': 'This registration is already pending a retraction.'
+        })
+
+    return serialize_node(node, auth, primary=True)
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_be_public_registration
+def node_registration_retraction_post(auth, node, **kwargs):
+    """Handles retraction of public registrations
+
+    :param auth: Authentication object for User
+    :return: Redirect URL for successful POST
+    """
+
+    data = request.get_json()
+    try:
+        node.retract_registration(auth.user, data.get('justification', None))
+        node.add_log(
+            action=NodeLog.RETRACTION_INITIATED,
+            params={
+                'registration_id': node._id,
+                'retraction_id': node.retraction._id,
+            },
+            auth=auth,
+            log_date=datetime.datetime.utcnow(),
+            save=False,
+        )
+        node.save()
+    except ValidationValueError as err:
```

Comment:
I'm not seeing where `retract_registration` can raise a `ValidationValueError`; is this supposed to be `NodeStateError`? Do both error types need to be handled?